prboom2/src/dsda/dgeom.h

```diff
@@ -0,0 +1,212 @@
+//
+// Copyright(C) 2023 Brian Koropoff
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// DESCRIPTION:
+//	Internal double-precision geometry functions
+//
+//      Not intended for use outside of DSDA subdir, so these aren't
+//      prefixed with dsda_
+//
+
+#ifndef DSDA_INTERNAL_
+#error Internal header, not for use outside dsda subdir
+#endif
+
+#ifndef DSDA_DGEOM_H_
```

Comment:
Please use the existing header pattern, in this case `__DSDA_DGEOM__`

Reply:
FYI, all identifiers starting with `__` are reserved by the C standard for system headers and shouldn't be used in user programs.  C++ is even more strict, see https://en.cppreference.com/w/c/language/identifier

Reply:
I appreciate that but every header uses the underscores like this, and I have no interest in changing that right now 😄

Reply:
But, these aren't identifiers, so I'm not sure the doc you linked applies here. The defined strings are only seen by the compiler and don't have "meaning".

Reply:
Macro names are considered identifiers by the spec.  I changed this to follow the existing convention anyway.

Reply:
Oh I see. But why doesn't the compiler complain then?

Reply:
Clang will complain in some cases with `-Wreserved-identifier`, but compilers have generally not caught it because:

- System headers are allowed to use reserved identifiers, and the compiler can't easily tell what's a system header and what isn't.
- It's undefined behavior according to the spec, which means it's perfectly fine for your program to subtly stop working on the next version of Windows or MSVC or whatever because you used a reserved identifier.

Reply:
I see, thanks for the info 🙌